lib/sgtn-client/util/locale-util.rb

```diff
@@ -10,10 +15,29 @@ module SgtnClient
                   }
 
     class LocaleUtil
+
+
+        def self.get_best_locale(locale)
+            flocale = fallback(process_locale(locale))
+            flocale = SgtnClient::Config.configurations.default if flocale == get_source_locale()
+            return flocale
+        end
+
         def self.process_locale(locale=nil)
             locale ||= SgtnClient::Config.configurations.default
             locale.to_s
         end
+
+        def self.get_source_locale
+            env = SgtnClient::Config.default_environment
+            default_l = SgtnClient::Config.configurations[env]["default_language"]
+            default_l || 'en'
+        end
+
+        def self.is_default(locale=nil)
+            locale == get_source_locale
```

Comment:
default maybe different from source_locale

Reply:
I changed the word to avoid misunderstanding, it meant source_locale actually.